docs-ui/components/code.tsx

```diff
@@ -54,11 +55,20 @@ const Code = ({children, className, label}: Props) => {
 
     setCopied(true);
 
-    setTimeout(() => {
+    copyTimeoutRef.current = window.setTimeout(() => {
       setCopied(false);
     }, 500);
   }
 
+  // Cleanup timeout on component unmount
+  useEffect(() => {
+    return () => {
+      if (copyTimeoutRef.current !== null) {
+        window.clearTimeout();
```

Comment:
Shouldn't we pass the ref?
```suggestion
        window.clearTimeout(copyTimeoutRef.current);
```